moveit_simple/src/robot.cpp

```diff
@@ -0,0 +1,1606 @@
+                                       << time);
+
+  try
+  {
+    Eigen::Affine3d pose_rel_robot = transformToBase(pose, frame);
+    std::unique_ptr<TrajectoryPoint> point =
+        std::unique_ptr<TrajectoryPoint>(new CartTrajectoryPoint(pose_rel_robot, time, point_name));
+    addTrajPoint(traj_name, point, type, num_steps);
+  }
+  catch (tf2::TransformException& ex)
+  {
+    ROS_ERROR_STREAM("Add to trajectory failed for arbitrary pose point: " << ex.what());
+    throw ex;
+  }
+}
+
+void Robot::addTrajPoint(const std::string& traj_name, const Eigen::Affine3d pose, const std::string& pose_frame,
+                         const std::string& tool_name, double time, const InterpolationType& type,
+                         const unsigned int num_steps, const std::string& point_name)
+{
+  std::lock_guard<std::recursive_mutex> guard(m_);
+
+  auto moveit_tool_link = joint_group_->getSolverInstance()->getTipFrame();
+
+  ROS_INFO_STREAM("Attempting to add " << point_name << " to " << traj_name << " relative to " << pose_frame
+                                       << " at time " << time << " and custom_tool_frame [" << tool_name << "]");
+
+  try
+  {
+    ROS_INFO_STREAM("Transforming Pose to custom_tool_frame frame [" << tool_name << "] from moveit_end_link ["
+                                                                     << moveit_tool_link << "]");
+
+    auto pose_rel_robot = this->transformToBase(pose, pose_frame);
+    auto custom_tool_to_moveit_tool = this->lookupTransformMoveitToolAndCustomTool(tool_name);
+    Eigen::Affine3d custom_tool_to_moveit_tool_eigen;
+    tf::transformMsgToEigen(custom_tool_to_moveit_tool.transform, custom_tool_to_moveit_tool_eigen);
+    pose_rel_robot = pose_rel_robot * custom_tool_to_moveit_tool_eigen;
+
+    std::unique_ptr<TrajectoryPoint> point =
+        std::unique_ptr<TrajectoryPoint>(new CartTrajectoryPoint(pose_rel_robot, time, point_name));
+
+    addTrajPoint(traj_name, point, type, num_steps);
+  }
+  catch (tf2::TransformException& ex)
+  {
+    ROS_WARN_STREAM("Add to trajectory failed for arbitrary pose point in Frame[" << tool_name << "]: " << ex.what());
+    throw ex;
+  }
+}
+
+void Robot::addTrajPoint(const std::string& traj_name, const std::string& point_name, double time,
+                         const InterpolationType& type, const unsigned int num_steps)
+{
+  std::lock_guard<std::recursive_mutex> guard(m_);
+
+  ROS_INFO_STREAM("Attempting to add " << point_name << " to " << traj_name << " at time " << time);
+
+  try
+  {
+    std::unique_ptr<TrajectoryPoint> point = lookupTrajectoryPoint(point_name, time);
+    addTrajPoint(traj_name, point, type, num_steps);
+  }
+  catch (std::invalid_argument& ia)
+  {
+    ROS_ERROR_STREAM("Invalid point " << point_name << " to add to " << traj_name);
+    throw ia;
+  }
+  catch (tf2::TransformException& ex)
+  {
+    ROS_ERROR_STREAM(" TF transform failed for " << point_name << " to add to " << traj_name);
```

Comment:
Since this is sometimes handled upstream, I would recommend adding any additional information to the `errorDescription` rather than printing to ROS_ERROR_STREAM.